src/rebalancers/Rebalancer.sol

```diff
@@ -0,0 +1,492 @@
+     * relative to the price calculated with trusted external prices of both assets, with 18 decimals precision.
+     * @param fee The fee paid to to the initiator, with 18 decimals precision.
+     * @param minLiquidityRatio The ratio of the minimum amount of liquidity that must be minted,
+     * relative to the hypothetical amount of liquidity when we rebalance without slippage, with 18 decimals precision.
+     * @dev The tolerance for the pool price will be converted to an upper and lower max sqrtPrice deviation,
+     * using the square root of the basis (one with 18 decimals precision) +- tolerance (18 decimals precision).
+     * The tolerance boundaries are symmetric around the price, but taking the square root will result in a different
+     * allowed deviation of the sqrtPriceX96 for the lower and upper boundaries.
+     */
+    function setInitiatorInfo(uint256 tolerance, uint256 fee, uint256 minLiquidityRatio) external {
+        if (account != address(0)) revert Reentered();
+
+        // Cache struct
+        InitiatorInfo memory initiatorInfo_ = initiatorInfo[msg.sender];
+
+        // Calculation required for checks.
+        uint64 upperSqrtPriceDeviation = uint64(FixedPointMathLib.sqrt((1e18 + tolerance) * 1e18));
+
+        // Check if initiator is already set.
+        if (initiatorInfo_.minLiquidityRatio > 0) {
+            // If so, the initiator can only change parameters to more favourable values for users.
+            if (
+                fee > initiatorInfo_.fee || upperSqrtPriceDeviation > initiatorInfo_.upperSqrtPriceDeviation
+                    || minLiquidityRatio < initiatorInfo_.minLiquidityRatio || minLiquidityRatio > 1e18
+            ) revert InvalidValue();
+        } else {
+            // If not, the parameters can not exceed certain thresholds.
+            if (
+                fee > MAX_INITIATOR_FEE || tolerance > MAX_TOLERANCE || minLiquidityRatio < MIN_LIQUIDITY_RATIO
+                    || minLiquidityRatio > 1e18
+            ) {
+                revert InvalidValue();
+            }
+        }
+
+        initiatorInfo_.fee = uint64(fee);
+        initiatorInfo_.minLiquidityRatio = uint64(minLiquidityRatio);
+        initiatorInfo_.lowerSqrtPriceDeviation = uint64(FixedPointMathLib.sqrt((1e18 - tolerance) * 1e18));
+        initiatorInfo_.upperSqrtPriceDeviation = upperSqrtPriceDeviation;
+
+        initiatorInfo[msg.sender] = initiatorInfo_;
+    }
+
+    /* ///////////////////////////////////////////////////////////////
+                      ERC721 HANDLER FUNCTION
+    /////////////////////////////////////////////////////////////// */
+
+    /**
+     * @notice Returns the onERC721Received selector.
+     * @dev Required to receive ERC721 tokens via safeTransferFrom.
+     */
+    function onERC721Received(address, address, uint256, bytes calldata) public pure returns (bytes4) {
+        return this.onERC721Received.selector;
+    }
+}
```

Comment:
should we add a fallback revert? or does it expect to have eth at some point?

Reply:
Good catch, we need a non reverting receive, since Slipstream Position manager can send funds.
Similar as we have it in the Slipstream compounder